kong/plugins/rate-limiting/handler.lua

```diff
@@ -105,25 +111,61 @@ function RateLimitingHandler:access(conf)
     end
   end
 
+  local timestamps
+  local reset
   if usage then
     -- Adding headers
     if not conf.hide_client_headers then
       local headers = {}
+      -- Find the closer limit.
+      local lowest = nil
       for k, v in pairs(usage) do
         if stop == nil or stop == k then
           v.remaining = v.remaining - 1
         end
 
+        local remaining = max(0, v.remaining)
+
         headers[RATELIMIT_LIMIT .. "-" .. k] = v.limit
-        headers[RATELIMIT_REMAINING .. "-" .. k] = max(0, v.remaining)
+        headers[RATELIMIT_REMAINING .. "-" .. k] = remaining
+
+        if stop and remaining == 0 then
+          if not timestamps then
+            timestamps = timestamp.get_timestamps(current_timestamp)
+          end
+
+          reset = max(reset or 0, EXPIRATIONS[k] - floor((current_timestamp -
+                                                          timestamps[k]) / 1000))
+        end
+
+        -- find the lowest limit in terms of remaining calls or time-window
+        if (lowest == nil
+            or usage[lowest].remaining > remaining
+            or (usage[lowest].remaining == remaining and EXPIRATIONS[k] > EXPIRATIONS[lowest])
+           ) then
+          lowest = k
+        end
       end
 
+      -- Set the normalized ratelimit values.
+      headers[RATELIMIT_LIMIT] = headers[RATELIMIT_LIMIT .. "-" .. lowest]
```

Comment:
Perhaps the generic headers should follow this (as they are new, without `X-` prefix):
https://tools.ietf.org/html/draft-polli-ratelimit-headers-01#section-6.2.4

Reply:
@bungle yes, but we are allowed to remove the `X-` only after the draft becomes RFC.

Reply:
@ioggstream, AFAIK this:
https://tools.ietf.org/html/rfc6648

has deprecated the use of `X-` in 2012?

Reply:
Yes, but using `DraftPolliV1-RateLimit-Limit` was weird :)))

Reply:
Perhaps we can take a risk and just use `RateLimit-Limit`, if the RFC changes it, we should change as well. But still keeping these old `X-` headers that we have currently. We can document that those generic headers follow the RFC and if RFC is changed, we will change too.

Reply:
Seems good. Are you interested in contributing to the spec? In case write me at robipolli@gmail.com